crates/project/src/terminals.rs

```diff
@@ -404,31 +404,40 @@ impl Project {
         &mut self,
         terminal: &Entity<Terminal>,
         cx: &mut Context<'_, Project>,
-        cwd: impl FnOnce() -> Option<PathBuf>,
+        cwd: Option<PathBuf>,
     ) -> Result<Entity<Terminal>> {
-        terminal.read(cx).clone_builder(cx, cwd).map(|builder| {
-            let terminal_handle = cx.new(|cx| builder.subscribe(cx));
-
-            self.terminals
-                .local_handles
-                .push(terminal_handle.downgrade());
-
-            let id = terminal_handle.entity_id();
-            cx.observe_release(&terminal_handle, move |project, _terminal, cx| {
-                let handles = &mut project.terminals.local_handles;
-
-                if let Some(index) = handles
-                    .iter()
-                    .position(|terminal| terminal.entity_id() == id)
-                {
-                    handles.remove(index);
-                    cx.notify();
-                }
-            })
-            .detach();
+        let local_path = if self.is_via_remote_server() {
+            None
+        } else {
+            cwd
+        };
```

Comment:
This I think is the wrong place to branch, we should check for whether the terminal we spawn is remote and apply the cwd correctly there. That is, if the terminal we spawn is a remote one, we want to use the cwd for the actual shell spawned in the remote, not for the `ssh`/`wsl` command

Reply:
Ah okay that makes sense